.gitattributes

```diff
@@ -0,0 +1,7 @@
+# Automatically normalize line endings
+* text=auto
+
+# Always use lf for F# files
+*.fs text eol=lf
+*.fsx text eol=lf
+*.fsi text eol=lf
```

Comment:
why this instead of just relying on auto? does setting this require a re-indexing of the existing files in the repo?

Reply:
This is something that @nojaf added, and seems to do in a few of their repos. Perhaps he could explain? :)

Reply:
It's generally less of a headache if all F# files are treated as LF. Setting this will override any global settings users may have. We've encountered issues in the past where editors flagged all files as changed after formatting because users checked them out on Windows, even with `end_of_line = lf` in `.editorconfig`.

Reply:
Hah, after changing branch (or some other action!), it's applied to my files and now tests are failing locally as I'm comparing triple-quoted strings with LF against the CRLF from the SDK. So I'll need to update that!